src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -864,6 +867,11 @@ bool RTPSParticipantImpl::create_reader(
         return false;
     }
 
+    // Use participant's external locators if reader has none
+    // WARNING: call before createAndAssociateReceiverswithEndpoint, as the latter intentionally clears external
+    // locators list when using unique_flows feature
```

Comment:
I think we should also put this warning on the `create_writer` method above

Reply:
I intentionally did not add it because ``unique_flows`` argument is only passed to ``createAndAssociateReceiverswithEndpoint`` in ``create_reader``.